### PR TITLE
SAK-43946: Lessons-TQ: cc_import skips or does not import some questions correctly

### PR DESCRIPTION
**Pull request by jesusmmp**
This one only solves: **Numeric Response: turned this into a Fill in the Blank question**
from: https://jira.sakaiproject.org/browse/SAK-43946.

WARNING: ¿Should be this one the new behaviour?
There is no option to Sakai from Export CC

image

If this is merged is possible than then works on Sakai but not on anothers LMS: Moodle, BlackBoard, Canvas, etc.

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 1]
```diff
@@ -1078,7 +1101,11 @@ boolean procshort(Node item) throws IOException{
 	out.println("    <qtimetadata>");
 	out.println("      <qtimetadatafield>");
 	out.println("        <fieldlabel>qmd_itemtype</fieldlabel>");
-	out.println("        <fieldentry>Fill In the Blank</fieldentry>");
+	if (type.equalsIgnoreCase("Fill In the Blank")) {
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:1104]: IMO, "Fill In the Blank" should be defined in a const somewhere. This string is all over the place in the Samigo and Lessons code and you would be doing the community a service by cleaning those up.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 2]
```diff
@@ -181,7 +181,21 @@ public String guessQuestionType (Node itemnode) {
...
 	}
-	
+
+	// response_fin is for numeric response
+	// matching has an explicit type, so just need to see if true false
+	NodeList fin=((Element)itemnode).getElementsByTagName("render_fin");
+	if (fin != null && fin.item(0) != null) {
+
+	    NodeList varequal=((Element)itemnode).getElementsByTagName("varequal");
+	    if (varequal != null && varequal.item(0) != null)
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:191]: Need braces around these ifs.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 3]
```diff
@@ -181,7 +181,21 @@ public String guessQuestionType (Node itemnode) {
...
+		return "Numeric Response";
+	    NodeList varsubstring=((Element)itemnode).getElementsByTagName("varsubstring");
+	    if (varsubstring != null && varsubstring.item(0) != null)
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:194]: Braces around the ifs make for less mistakes down the line.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 4]
```diff
@@ -1087,13 +1114,15 @@ boolean procshort(Node item) throws IOException{
 	out.println("        <fieldentry>HTML</fieldentry>");
 	out.println("      </qtimetadatafield>");
 	out.println("    </qtimetadata>");
-	out.println("    <qtimetadata>");
-	out.println("      <qtimetadatafield>");
-	out.println("        <fieldlabel>CASE_SENSITIVE</fieldlabel>");
-	out.println("        <fieldentry>"+casesens+"</fieldentry>");
-	out.println("      </qtimetadatafield>");
+    out.println("    <qtimetadata>");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:1117]: Indentation

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 5]
```diff
@@ -1128,7 +1157,11 @@ boolean procshort(Node item) throws IOException{
 	    if (ident == null)
 		ident = ans.qident;
 
-	    out.println("<response_str ident=\""+rident+"\" rcardinality=\"Ordered\" rtiming=\"No\"><render_fib charset=\"ascii-us\" columns=\"20\" encoding=\"UTF_8\" fibtype=\"String\" prompt=\"Box\" rows=\"1\"></render_fib>");
+	    if (type.equalsIgnoreCase("Fill In the Blank")) {
+		    out.println("<response_str ident=\""+rident+"\" rcardinality=\"Ordered\" rtiming=\"No\"><render_fib charset=\"ascii-us\" columns=\"20\" encoding=\"UTF_8\" fibtype=\"String\" prompt=\"Box\" rows=\"1\"></render_fib>");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:1161]: My preference is to use spaces around + signs in concatenation, and this code uses that approach elsewhere. It just makes it easier to read.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 6]
```diff
@@ -1128,7 +1157,11 @@ boolean procshort(Node item) throws IOException{
...
+	    } else if (type.equalsIgnoreCase("Numeric Response")) {
+		    out.println("<response_str ident=\""+rident+"\" rcardinality=\"Ordered\" rtiming=\"No\"><render_fin charset=\"ascii-us\" columns=\"20\" encoding=\"UTF_8\" fibtype=\"String\" prompt=\"Box\" rows=\"1\"></render_fin>");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:1163]: Same here. Spaces around the + signs, easier on the eye.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 7]
```diff
@@ -181,7 +181,21 @@ public String guessQuestionType (Node itemnode) {
 		return "Fill In the Blank";
 
 	}
-	
+
+	// response_fin is for numeric response
+	// matching has an explicit type, so just need to see if true false
+	NodeList fin=((Element)itemnode).getElementsByTagName("render_fin");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:187]: I'd use spaces around the = sign here. Easier to read.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 8]
```diff
@@ -181,7 +181,21 @@ public String guessQuestionType (Node itemnode) {
...
+	if (fin != null && fin.item(0) != null) {
+
+	    NodeList varequal=((Element)itemnode).getElementsByTagName("varequal");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:190]: spaces around the = sign

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java [hunk 9]
```diff
@@ -181,7 +181,21 @@ public String guessQuestionType (Node itemnode) {
...
+	    if (varequal != null && varequal.item(0) != null)
+		return "Numeric Response";
+	    NodeList varsubstring=((Element)itemnode).getElementsByTagName("varsubstring");
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:193]: spaces around the = sign

**Review by adrianfish — CHANGES_REQUESTED** [review]

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 2 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 3 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 4 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 5 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 6 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 7 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 8 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 9 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java]: Solved

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java (hunk 1 above)
Comment (jesusmmp) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/QtiImport.java:1104]: @adrianfish, I'm agree with you but these 2 classes (QtiImport.java and SamigoExport.java) are really a big disaster about: no const, no spacing, no braces, bad formatting, unnecessary comments etc.
IMO all that things should be resolved on a new one. Please, file a new jira for that purpose. Thanks!

**Comment by jesusmmp** [comment]
More worries about why there are 2 PRs instead of reading the description and it's content itself. I'm closing it. Thanks.

**Pull request closed without merging**